iceoryx_posh/test/moduletests/test_roudi_service_registry.cpp

```diff
@@ -445,4 +465,109 @@ TEST_F(ServiceRegistry_test, AddingVariousServiceDescriptionAndGetServicesDoesNo
     EXPECT_THAT(service1Found && service2Found && service4Found, Eq(true));
 }
 
+template <typename T>
+T uniform(T max)
+{
+    static auto seed = std::random_device()();
+    static std::mt19937 mt(seed);
+    std::uniform_int_distribution<T> dist(0, max);
+    return dist(mt);
+}
+
+using string_t = iox::capro::IdString_t;
+
+string_t randomString(uint64_t size = string_t::capacity())
+{
+    // deliberately contains no `0` (need to exclude some char)
+    static const char chars[] = "123456789"
+                                "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+                                "abcdefghijklmnopqrstuvwxyz";
+
+    constexpr auto N = string_t::capacity();
+    size = std::min(N, size);
+
+    char a[N + 1];
+    for (uint64_t i = 0; i < size; ++i)
+    {
+        a[i] = chars[uniform(sizeof(chars) - 1)];
+    }
+    a[size] = '\0';
+
+    string_t s(a);
+    return s;
+}
+
+TEST_F(ServiceRegistry_test, CanAddMaximumNumberOfDifferentServiceDescriptions)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "76aef6cb-7886-4d64-9188-09bd1be2d335");
+    uint32_t numEntriesAdded = 0;
+    do
+    {
+        // may (rarely) generate duplicates to be counted internally
+        auto id = randomString();
+        ServiceDescription sd(id, id, id);
+        auto result = sut.add(sd);
+        if (result.has_error())
+        {
+            break;
+        }
+        numEntriesAdded++;
+    } while (true);
+
+    // duplicates do not count to the max and may be generated randomly,
+    // but for the contract we only need to guarantee that we can at least add
+    // the configured max
+    constexpr auto MAX = ServiceRegistry::MAX_SERVICE_DESCRIPTIONS;
+    EXPECT_GE(numEntriesAdded, MAX);
+}
+
+TEST_F(ServiceRegistry_test, SearchInFullRegistryWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "5c4519d4-1873-4837-a3eb-0367106fb9b5");
+
+    constexpr auto CAP = string_t::capacity();
+
+    string_t fixedId(iox::cxx::TruncateToCapacity, std::string(CAP, '0'));
+
+    ServiceDescription lastAdded;
+    do
+    {
+        auto id = randomString();
+        ServiceDescription sd(fixedId, fixedId, id);
+
+        auto result = sut.add(sd);
+        if (result.has_error())
+        {
+            break;
+        }
+        lastAdded = sd;
```

Comment:
Needs to be stored for removal afterwards since we do not know when the registry is full if we add random values (duplicates occupy no extra space).